src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -3227,3 +3187,15 @@ void pushNPCSpell(SPELL_TYPE spell) {
 void pushScrollSpell(SPELL_TYPE spell, unsigned int uPlayerID) {
     pushSpellOrRangedAttack(spell, uPlayerID, SCROLL_OR_NPC_SPELL_SKILL_VALUE, ON_CAST_CastViaScroll, 0);
 }
+
+void spellTargetPicked(int pid, int playerTarget) {
+    assert(pid != PID_INVALID ^ playerTarget != -1);
```

Comment:
I believe this won't work.

https://en.cppreference.com/w/cpp/language/operator_precedence

You're XORing PID_INVALID and playerTarget. Need parens around the comparisons.